flink-core/src/main/java/org/apache/flink/configuration/ExecutionOptions.java

```diff
@@ -92,6 +92,17 @@ public class ExecutionOptions {
                     .withDescription(
                             "Tells if we should use compression for the state snapshot data or not");
 
+    public static final ConfigOption<Boolean> BUFFER_TIMEOUT_ENABLED =
+            ConfigOptions.key("execution.buffer-timeout.enabled")
```

Comment:
FLINK-29372 limited that the option key cannot be a prefix of other options, however `execution.buffer-timeout` is the prefix of `execution.buffer-timeout.enabled`. So the CI fails.

Hi @zentol , could we update the option key for `BUFFER_TIMEOUT` and mark the `"execution.buffer-timeout"` as the DeprecatedKey? How about update it to `execution.buffer-timeout.interval` ? Or do you have any suggestions here? Looking forward to your opinion.

Reply:
How about changing the config execution.buffer-timeout.enabled to execution.flush-on-buffer-full?

Reply:
Hi @Myracle , thanks for your feedback.

I see too many related option names that all have the same prefix but different suffixes. From the user's point of view, the same prefix also clearly tells the user that these options are related. Therefore, when FLINK-29372 restricts the option prefix, it also adds suffixes to many option names that do not conform to the specification.

So I prefer keep the prefixes the same for these 2 options. In order to the reasonable of this fix, I look forward to more feedback from community.

Hi @wanglijie95  @reswqa , would you mind take a look this PR in your free time? thanks a lot.

Reply:
Rename it to `execution.buffer-timeout.interval` and mark the old as deprecated key is ok from my side. But I want to hear @zentol's thoughts.

Reply:
BTW, I'm not sure do we need to start a discussion about this changes of configuration in the mailing list.

Reply:
+1 for name `execution.buffer-timeout.interval`

Reply:
Rename it to execution.buffer-timeout.interval. @zentol Would you like to give your thought? Thanks.

Reply:
`execution.buffer-timeout.interval` and `execution.buffer-timeout.enabled` sound fine to me.